Extensions/BBText/bbtextruntimeobject.js

```diff
@@ -0,0 +1,206 @@
+/**
+ * An object displaying a BBCode formated text on the screen.
+ * @memberof gdjs
+ * @class BBTextRuntimeObject
+ * @extends RuntimeObject
+ */
+gdjs.BBTextRuntimeObject = function(runtimeScene, objectData) {
+  gdjs.RuntimeObject.call(this, runtimeScene, objectData);
+
+  /** @type number */
+  this._opacity = objectData.content.opacity;
+  /** @type boolean */
+  this._visible = objectData.content.visible;
+  /** @type string */
+  this._text = objectData.content.text;
+  /** @type string */
+  this._color = objectData.content.color;
+  /** @type string */
+  this._family = objectData.content.family;
+  /** @type number */
+  this._size = objectData.content.size;
+  /** @type number */
+  this._wrappingWidth = objectData.content.width;
+  /** @type string */
+  this._align = objectData.content.align;
+
+  if (this._renderer)
+    gdjs.BBTextRuntimeObjectRenderer.call(this._renderer, this, runtimeScene);
+  else
+    this._renderer = new gdjs.BBTextRuntimeObjectRenderer(this, runtimeScene);
+
+  // *ALWAYS* call `this.onCreated()` at the very end of your object constructor.
+  this.onCreated();
+};
+
+gdjs.BBTextRuntimeObject.prototype = Object.create(
+  gdjs.RuntimeObject.prototype
+);
+gdjs.BBTextRuntimeObject.thisIsARuntimeObjectConstructor = 'BBText::BBText';
+
+gdjs.BBTextRuntimeObject.prototype.getRendererObject = function() {
+  return this._renderer.getRendererObject();
+};
+
+/**
+ * Initialize the extra parameters that could be set for an instance.
+ * @private
+ */
+gdjs.BBTextRuntimeObject.prototype.extraInitializationFromInitialInstance = function(
+  initialInstanceData
+) {
+  if (initialInstanceData.customSize) {
+    this.setWidth(initialInstanceData.width);
+    this.setWrappingWidth(initialInstanceData.width);
+  }
+};
+
+gdjs.BBTextRuntimeObject.prototype.onDestroyFromScene = function(runtimeScene) {
+  gdjs.RuntimeObject.prototype.onDestroyFromScene.call(this, runtimeScene);
+
+  this._renderer.onDestroy();
+};
+
+gdjs.BBTextRuntimeObject.prototype.update = function(runtimeScene) {
+  if (this._renderer) this._renderer.ensureUpToDate();
+};
+
+/**
+ * Set/Get BBText base style properties
+ */
+gdjs.BBTextRuntimeObject.prototype.setBBText = function(text) {
+  this._text = text;
+  this._renderer.updateText();
+};
+
+gdjs.BBTextRuntimeObject.prototype.getBBText = function(text) {
+  return this._text;
+};
+
+gdjs.BBTextRuntimeObject.prototype.setColor = function(value) {
+  const splitValue = value.split(';');
+  const hexColor = `#${gdjs.rgbToHex(
+    parseInt(splitValue[0]),
+    parseInt(splitValue[1]),
+    parseInt(splitValue[2])
+  )}`;
+  this._color = hexColor;
+  this._renderer.updateColor();
+  this.update();
+};
+
+gdjs.BBTextRuntimeObject.prototype.getColor = function() {
+  return this._color;
+};
+
+gdjs.BBTextRuntimeObject.prototype.setFontSize = function(value) {
+  this._size = `${value}px`;
```

Comment:
You declared earlier that size is a number, but you're setting a string here.
Remove the string interpolation `${value}px` and add instead "px" in the renderer.